packages/dates/src/parse-date-string.ts

```diff
@@ -3,49 +3,60 @@ import {applyTimeZoneOffset} from './apply-time-zone-offset';
 /**
  * Allowed date string formats
  * yyyy-mm-dd
- * yyyy-mm-ddThh:mm:ss
- * yyyy-mm-ddThh:mm:ss+hh:mm
- * yyyy-mm-ddThh:mm:ss-hh:mm
- * yyyy-mm-ddThh:mm:ssZ
+ * yyyy-mm-ddThh:mm:ss.fff
+ * yyyy-mm-ddThh:mm:ss.fff+hh:mm
+ * yyyy-mm-ddThh:mm:ss.fff-hh:mm
+ * yyyy-mm-ddThh:mm:ss.fffZ
  */
-const DATE_TIME_PARTS_REGEX = /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}):(\d{2}):(\d{2})(?:(Z|(?:(\+|-)(\d{2}):(\d{2}))))?)?$/;
+const DATE_TIME_PARTS_REGEX = /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}):(\d{2}):(\d{2})(?:\.(\d{1,3}))?(?:(Z|(?:(\+|-)(\d{2}):(\d{2}))))?)?$/;
 
-export function parseDateString(dateString: string, timeZone?: string) {
+export function parseDateStringParts(dateString: string) {
   const dateTimeParts = new RegExp(DATE_TIME_PARTS_REGEX).exec(dateString);
 
   if (dateTimeParts == null) {
     return null;
   }
 
+  // slice the first regex part (the full match) off
+  return Array.from(dateTimeParts).slice(1);
```

Comment:
I always feel a bit uncomfortable when a function returns an array, it might be better to move the destructuring and `parseInt`'ing into this function, and return an object with all of the 'parts'.

That also allows us to surface a `DateStringParts` interface in the API

Reply:
I would be fine with that. i'll make the change.